docs/tutorial.md

```diff
@@ -0,0 +1,369 @@
+NewCreateAPIScaffolder called
+InjectFS called
+api.Scaffold()
+PostScaffold called
+...
+```
+
+This will scaffold the `MemcachedController`, `MemcachedSpec`, `MemcachedStatus` and `Memcached`.
+
+#### Understanding Kubernetes APIs
+
+For an in-depth explanation of Kubernetes APIs and the group-version-kind model, check out these [kubebuilder docs](https://book.kubebuilder.io/cronjob-tutorial/gvks.html).
+
+The `java-operator-plugins` project uses the APIs from [java-operator-sdk](https://github.com/java-operator-sdk/java-operator-sdk). The java-operator-sdk library is to Java what `controller-runtime` is to Go. It provides a framework to help Java projects interact with the Kubernetes API.
+
+### Define the API
+
+`create api` command will scaffold out an `MemcachedController`, `MemcachedSpec`, `MemcachedStatus` and `Memcached`. 
+
+#### `MemcachedSpec`
+
+Initially, the scaffolded Spec file will be empty. The operator developer needs to add attributes to this file according to his need. For the Memcached example, we added the size field as shown below example.
+
+```
+// MemcachedSpec defines the desired state of Memcached
+public class MemcachedSpec {
+
+    // Add Spec information here
+	// Size is the size of the memcached deployment
+    private Integer size;
+
+    public Integer getSize() {
+        return size;
+    }
+
+    public void setSize(Integer size) {
+        this.size = size;
+    }
+}
+```
+
+#### `MemcachedStatus`
+
+Similar to the Spec file `MemcachedStatus` file got scaffolded as part of the `create api` command. The user has to modify a Status file. For the Memcached example, we too list of nodes as shown below.
+The nodes field is a list of string values and it contains the name of the Memcached pods.
+
+
+```
+import java.util.ArrayList;
+import java.util.List;
+
+// MemcachedStatus defines the observed state of Memcached
+public class MemcachedStatus {
+
+    // Add Status information here
+	// Nodes are the names of the memcached pods
+    private List<String> nodes;
+
+    public List<String> getNodes() {
+        if (nodes == null) {
+            nodes = new ArrayList<>();
+        }
+        return nodes;
+    }
+
+    public void setNodes(List<String> nodes) {
+        this.nodes = nodes;
+    }
+}
+```
+
+**Note** The Node field is just to illustrate an example of a Status field. In real cases, it would be recommended to use [Conditions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties).
+
+#### `Memcached`
+
+`Memcached` file scaffolded via `create api` command. It extends the properties of `MemcachedSpec` and `MemcachedStatus`.
+
+```
+// Memcached is the Schema for the memcacheds API
+
+@Version("v1alpha1")
+@Group("cache.example.com")
+public class Memcached extends CustomResource<MemcachedSpec, MemcachedStatus>
+    implements Namespaced {}
+```
+
+### Apply CR and CRD's using below command
+
+#### CRD - Custom Resource
```

Comment:
I found "CR" in `<h3>` and "Custom Resource" in `<h4>` a bit confusing.
I know we had this "CRD.yaml defines a CR" and "CRD is an endpoint" convo separately.

But I wonder if it'll be simpler if we just exclude "CR/Custom Resource" altogether from the `<h3>` and `<h4>` headers simply as: "Apply CRD using below command" and "CRD - Custom Resource Definition"?

What do you guys think? @laxmikantbpandhare @jmrodri ?

Reply:
Agree with you @tlwu2013. I will mentioned CR as Custom Resource and Custom Resource Definition as CRD. It will be uniform throughout the document.